src/Nethermind/Nethermind.Runner/Ethereum/Steps/StartRpc.cs

```diff
@@ -30,8 +32,34 @@ public StartRpc(INethermindApi api)
         public async Task Execute(CancellationToken cancellationToken)
         {
             IJsonRpcConfig jsonRpcConfig = _api.Config<IJsonRpcConfig>();
+            IKeyStoreConfig keyStoreConfig = _api.Config<IKeyStoreConfig>();
             ILogger logger = _api.LogManager.GetClassLogger();
 
+
+            // Update the JWT secret path based on the data directory.
+            try
+            {
+                string defaultPath = "keystore/jwt-secret";
+                string newPath = Path.Combine(keyStoreConfig.KeyStoreDirectory, "jwt-secret");
+                if (string.IsNullOrEmpty(jsonRpcConfig.JwtSecretFile))
+                {
+                    logger.Warn("JsonRpcConfig JwtSecretFile is null or empty because of omission or human error.");
+                    // check if jwt-secret file already exists in previous default directory
+                    if (File.Exists(defaultPath))
+                    {
+                        // move the jwt-secret file
+                        logger.Warn($"jwt-secret already exists at {defaultPath}. Moving it to {newPath} as data directory has been updated");
+                        File.Move(defaultPath, newPath);
+                    }
```

Comment:
We better check whether another JWT secret exists before moving. Also, we catch exceptions here if any:
```suggestion
                    if (!File.Exists(newPath) && File.Exists(defaultPath))
                    {
                        try
                        {
                            File.Move(defaultPath, newPath);

                            logger.Warn($"Moved JWT secret from {defaultPath} to {newPath}");
                        }
                        catch (Exception ex)
                        {
                            if (logger.IsError) logger.Error($"Failed moving JWT secret to {newPath}.", ex);
                        }
                    }
```